libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -926,8 +926,12 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         // Telemetry uses this field to determine what type of event should be triggered.
         val setRoutesInfo = when {
             routes.isEmpty() -> SetRoutes.CleanUp
-            routes.first() == directionsSession.routes.firstOrNull() ->
-                SetRoutes.Alternatives(initialLegIndex)
+            routes.first() == directionsSession.routes.firstOrNull() -> {
+                SetRoutes.Alternatives(initialLegIndex, isPrimaryRouteChanged = false)
+            }
+            routes.map { it.id }.sorted() == directionsSession.routes.map { it.id }.sorted() -> {
```

Comment:
That's not reason "alternatives". This reason is not only used when setting routes to Native Navigator, but also:
1. In telemetry;
2. As reason in RoutesUpdatedResult (public API);
3. In some internal logic (but looks like it is adjusted).

But now you're trying to set reason alternatives where from our perspective it's not actually "alternatives". The main problem is the misunderstanding between us and NN, do we know where that came from?
Anyway, even if we leave this misalignment, I'd suggest introducing a new internal reason (like "Reorder"). And it would get mapped to native "alternative" and our "new" reasons correspondingly. 

BTW, what if we had routes [A, B, C] and now we have [B, A]? For us it's reason "new". What about NN? What do they expect?

Reply:
> As reason in RoutesUpdatedResult (public API)

It affects only mapping to `RoutesUpdatedResult.RoutesUpdateReason` (we haven't introduced any new value for `RoutesUpdateReason`), public api is not broken.

> The main problem is the misunderstanding between us and NN, do we know where that came from?

This behavior is expected by the NN, and now it's consistent with the iOS Navigation SDK (I'll send you a link to a discussion).

> Anyway, even if we leave this misalignment, I'd suggest introducing a new internal reason (like "Reorder"). And it would get mapped to native "alternative" and our "new" reasons correspondingly.

Makes sense, thank you!. I've updated the PR.

> BTW, what if we had routes [A, B, C] and now we have [B, A]? For us it's reason "new". What about NN? What do they expect?

It's seems that the reason `new` is expected, but we can make sure with the NN team.